File: Python/List-2/centered_average.py
def centered_average(nums):
  # nums.sort()
  # arr = nums[1:-1]
  # n = len(arr)
  # if n % 2 == 0:
  #   return int(((arr[int((n-1)/2)] +
  #                 arr[int(n/2)])/2.0))
  # return arr[n // 2]

  sum = 0
  for element in nums:
    sum += element
  return (sum - min(nums) - max(nums)) / (len(nums)-2)

File: Java/Array-2/centeredAverage.java
public int centeredAverage(int[] nums) {
  int len = nums.length;
  int max = nums[0];
  int min = nums[0];
  int total = nums[0];
  // find the min, max of the arr
  for (int i = 1; i < len; i++)
  {
    total += nums[i];
    max = Math.max(max, nums[i]);
    min = Math.min(min, nums[i]);
  }
  return (total - min - max) / (len - 2);
}
